src/storage/flagged.rs

```diff
@@ -166,16 +201,37 @@ impl<C: Component, T: UnprotectedStorage<C>> UnprotectedStorage<C> for FlaggedSt
 
     unsafe fn get_mut(&mut self, id: Index) -> &mut C {
         // calling `.iter()` on an unconstrained mutable storage will flag everything
+        #[cfg(feature = "storage-event-control")]
+        {
+            if self.event_emission {
+                self.channel.single_write(ComponentEvent::Modified(id));
+            }
+        }
+        #[cfg(not(feature = "storage-event-control"))]
         self.channel.single_write(ComponentEvent::Modified(id));
         self.storage.get_mut(id)
     }
 
     unsafe fn insert(&mut self, id: Index, comp: C) {
+        #[cfg(feature = "storage-event-control")]
+        {
+            if self.event_emission {
+                self.channel.single_write(ComponentEvent::Inserted(id));
+            }
+        }
+        #[cfg(not(feature = "storage-event-control"))]
         self.channel.single_write(ComponentEvent::Inserted(id));
         self.storage.insert(id, comp);
     }
 
     unsafe fn remove(&mut self, id: Index) -> C {
+        #[cfg(feature = "storage-event-control")]
```

Comment:
Can you please define a function which just returns `true` in case of a disabled feature?